test/integration/auth/svcaccttoken_test.go

```diff
@@ -196,9 +199,16 @@ func TestServiceAccountTokenCreate(t *testing.T) {
 			t.Fatalf("expected err creating token for nonexistant svcacct but got: %#v", resp)
 		}
 		warningHandler.assertEqual(t, nil)
+
+		// test that the warning is emitted when the service account has this deprecated annotation
+		sa.Annotations = map[string]string{EnforceMountableSecretsAnnotation: "true"}
 		sa, delSvcAcct := createDeleteSvcAcct(t, cs, sa)
 		defer delSvcAcct()
+		// restore the annotations to nil
+		sa.Annotations = map[string]string{}
+		warningHandler.assertEqual(t, []string{fmt.Sprintf("deprecated annotation %q on service account %s/%s", EnforceMountableSecretsAnnotation, sa.Namespace, sa.Name)})
 
+		warningHandler.clear()
```

Comment:
Use a new, distinct SA for the warning test, and then drop the logic around restoring annotations.